examples/custom-node/src/evm/env.rs

```diff
@@ -34,63 +37,112 @@ impl revm::context::Transaction for CustomEvmTransaction {
         Self: 'a;
 
     fn tx_type(&self) -> u8 {
-        self.0.tx_type()
+        match self {
+            CustomEvmTransaction::Op(tx) => tx.tx_type(),
+            CustomEvmTransaction::Payment(tx) => tx.tx_type(),
+        }
     }
 
     fn caller(&self) -> Address {
-        self.0.caller()
+        match self {
+            CustomEvmTransaction::Op(tx) => tx.caller(),
+            CustomEvmTransaction::Payment(tx) => tx.caller(),
+        }
     }
 
     fn gas_limit(&self) -> u64 {
-        self.0.gas_limit()
+        match self {
+            CustomEvmTransaction::Op(tx) => tx.gas_limit(),
+            CustomEvmTransaction::Payment(tx) => tx.gas_limit(),
+        }
     }
 
     fn value(&self) -> U256 {
-        self.0.value()
+        match self {
+            CustomEvmTransaction::Op(tx) => tx.value(),
+            CustomEvmTransaction::Payment(tx) => tx.value(),
+        }
     }
 
     fn input(&self) -> &Bytes {
-        self.0.input()
+        match self {
+            CustomEvmTransaction::Op(tx) => tx.input(),
+            CustomEvmTransaction::Payment(tx) => tx.input(),
+        }
     }
 
     fn nonce(&self) -> u64 {
-        revm::context::Transaction::nonce(&self.0)
+        match self {
+            CustomEvmTransaction::Op(tx) => revm::context::Transaction::nonce(tx),
+            CustomEvmTransaction::Payment(tx) => revm::context::Transaction::nonce(tx),
+        }
     }
 
     fn kind(&self) -> TxKind {
-        self.0.kind()
+        match self {
+            CustomEvmTransaction::Op(tx) => tx.kind(),
+            CustomEvmTransaction::Payment(tx) => tx.kind(),
+        }
     }
 
     fn chain_id(&self) -> Option<u64> {
-        self.0.chain_id()
+        match self {
+            CustomEvmTransaction::Op(tx) => tx.chain_id(),
+            CustomEvmTransaction::Payment(tx) => tx.chain_id(),
+        }
     }
 
     fn gas_price(&self) -> u128 {
-        self.0.gas_price()
+        match self {
+            CustomEvmTransaction::Op(tx) => tx.gas_price(),
+            CustomEvmTransaction::Payment(tx) => tx.gas_price(),
+        }
     }
 
     fn access_list(&self) -> Option<impl Iterator<Item = Self::AccessListItem<'_>>> {
```

Comment:
-.-
yeah returning an impl in the trait is a bit horrible, not much we can do here really